### PR TITLE
feat: add Static Module

### PR DESCRIPTION
**Pull request by Inconnu08**
- adds Static module
- fixes typos
- adds errchecks

### CODE REVIEW

**Comment by YourTechBud** [comment]
Wow @Inconnu08n! Insane to see how you pulled off this complete feature all by yourself. Awesome. Will review it tomorrow.

**Comment by Inconnu08** [comment]
Haha, I am sorry for the delay. Was kinda busy these days but thank you @YourTechBud!

File: config/config.go [hunk 1]
```diff
@@ -93,3 +94,11 @@ type FileRule struct {
 	Prefix string           `json:"prefix" yaml:"prefix"`
 	Rule   map[string]*Rule `json:"rule" yaml:"rule"` // The key can be create, read, delete
 }
+
+// Static holds the config for the static files module
+type Static struct {
+	Enabled bool   `json:"enabled" yaml:"enabled"`
+	Path    string `json:"path" yaml:"path"`
+	UrlPrefix string `json:"url_prefix" yaml:"url_prefix"`
```
Comment (YourTechBud) [inline config/config.go]:
```suggestion
	URLPrefix string `json:"prefix" yaml:"prefix"`
```

File: config/template.go [hunk 2]
```diff
@@ -43,4 +43,9 @@ modules:
             rule: allow
           delete:
             rule: allow
+  static:
+    enabled: true
+    path: ./static
+	url_prefix: /static/
```
Comment (YourTechBud) [inline config/template.go]:
```suggestion
    prefix: /
```

File: config/template.go [hunk 3]
```diff
@@ -43,4 +43,9 @@ modules:
             rule: allow
           delete:
             rule: allow
+  static:
+    enabled: true
+    path: ./static
```
Comment (YourTechBud) [inline config/template.go]:
```suggestion
    path: ./public
```

File: modules/auth/auth.go [hunk 4]
```diff
@@ -92,7 +91,7 @@ func (m *Module) parseToken(token string) (map[string]interface{}, error) {
 	tokenObj, err := jwt.Parse(token, func(token *jwt.Token) (interface{}, error) {
 		// Don't forget to validate the alg is what you expect:
 		if token.Method.Alg() != jwt.SigningMethodHS256.Alg() {
-			return nil, errors.New("Invalid signing method type")
+			return nil, errors.New("invalid signing method type")
```
Comment (YourTechBud) [inline modules/auth/auth.go]:
```suggestion
			return nil, errors.New("Invalid signing method type")
```

File: modules/faas/http.go [hunk 5]
```diff
@@ -8,22 +8,21 @@ import (
...
-
 	"github.com/spaceuptech/space-cloud/modules/auth"
 )
 
 // HandleRequest creates a FaaS request endpoint
 func (m *Module) HandleRequest(auth *auth.Module) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 
-		// Return if the faas module is not enabled
+		// Return if the static module is not enabled
```
Comment (YourTechBud) [inline modules/faas/http.go]:
```suggestion
		// Return if the faas module is not enabled
```

File: modules/static/http.go [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+		// Return 404 if the static module is not enabled
+		if !m.isEnabled() {
+			w.WriteHeader(http.StatusNotFound)
+			_ = json.NewEncoder(w).Encode(map[string]string{"error": "This feature isn't enabled"})
+			return
+		}
+
+		staticRouter := mux.NewRouter().StrictSlash(true)
+		staticRouter.PathPrefix("/static/").Handler(http.FileServer(&SpaFileSystem{http.Dir(m.getDirPath())}))
```
Comment (YourTechBud) [inline modules/static/http.go]:
```suggestion
		staticRouter.PathPrefix(m.URLPrefix).Handler(http.FileServer(&SpaFileSystem{http.Dir(m.getDirPath())}))
```

File: modules/static/static.go [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+func (m *Module) SetConfig(s *config.Static) error {
+	m.Lock()
+	defer m.Unlock()
+
+	if s == nil || !s.Enabled {
+		m.Enabled = false
+		return nil
+	}
+
+	if s.Gzip {
```
Comment (YourTechBud) [inline modules/static/static.go]: Can we do something like this
```go
m.Gzip = s.Gzip
```

File: modules/static/static.go [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+		m.Gzip = true
+	}
+
+	if s.Path != "" {
```
Comment (YourTechBud) [inline modules/static/static.go]: Something like this here: 
```golang
m.Path = s.Path
if m.Path == "" {
  m.Path == "./public"
}
```

File: modules/static/static.go [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+		m.Path = s.Path
+	}
+
+	if s.UrlPrefix != "" {
```
Comment (YourTechBud) [inline modules/static/static.go]: Something like this here:
```golang
m.URLPrefix = s.URLPrefix
if m.URLPrefix == "" {
  m.URLPrefix == "/"
}
```

File: modules/auth/auth.go (hunk 4 above)
Comment (Inconnu08) [reply modules/auth/auth.go]: But Go conventions say that `error` strings should not be capitalized @YourTechBud

File: modules/faas/http.go (hunk 5 above)
Comment (Inconnu08) [reply modules/faas/http.go]: Opps, silly typo

File: modules/static/static.go (hunk 7 above)
Comment (Inconnu08) [reply modules/static/static.go]: sure

Comment (Inconnu08) [reply modules/static/static.go]: But what if the config was removed from the template. Would it work?

**Comment by Inconnu08** [comment]
Refactored @YourTechBud

File: modules/auth/auth.go (hunk 4 above)
Comment (YourTechBud) [reply modules/auth/auth.go]: Makes sense. Cool. Let's discard this one

**Comment by YourTechBud** [comment]
Awesome @Inconnu08. I'll quickly go through these changes tomorrow then merge this right in.

**Comment by YourTechBud** [comment]
Hey @Inconnu08, im sorry i haven't been able to review this yet. Will take me a few weeks more to get this done.

**Comment by Inconnu08** [comment]
@YourTechBud It's alright, take your time bro!

**Pull request merged**